Foam force fuck yeah!

Fayrik opened the pull request:
![So much fun](http://i.imgur.com/3nGPhTs.png)
![GUNS](http://i.imgur.com/gD7nlaG.png)
I believe these images should speak for themselves.

Yeah, I've probably fucked some stuff up, but as far as I've tested, everything looks to be working correctly. If nothing else it's fun for a test server.
-- Important note: Carrying a toy gun will cause Securitrons with weapons check enabled to arrest you. After much deliberation, this is intended behaviour.
_Edit: I caved. They're no longer illegal._

So, some stuff added here:
- 4 new guns. A crossbow, shotgun, pistol and SMG.
- Crossbows are won from arcade machines, shotguns are normally available to cargo, pistols are available as contraband and SMGs are an admin weapon.
- All ammunition can be picked up into their appropriate ammo boxes and/or magazines instantly, by clicking them from the floor.
- Admin only foam darts that hit like disabler beams.

Fayrik committed:
First (of probably several) commits to add Foam Force toy guns. Adds the new guns and crossbow, removes the old crossbow, and makes these new guns available to cargo.

Fayrik committed:
Glorious second commit, introducing (mildly) dangerous rounds and a new suicide check.

Fayrik committed:
Removed dangerous "hacked" foam darts, as they were a little more powerful than intended. Hopefully this is something I can fix at a later date. Fixed non-magazine ammo boxes not accepting their correct ammo type. Added a speed collection system to all ammo boxes. Finally balanced out the cargo ordering stuff.

Fayrik committed:
Cleaned up the functionality and prepared things for PR.

RemieRichards commented:
>Important note: Carrying a toy gun will cause Securitrons with weapons check enabled to arrest you. After much deliberation, this is intended behaviour

Ewww, just eww.
There's an easy fix for this too.
ED209's and Securitrons have a ```safe_weapons``` list, which you'd just put the gun types in.
(Although if you WERE to use this I'd recommend replacing the list with a variable on the gun like, ```ignoreThreatAssess``` and have it default to 0, and set it on all your toy guns)

optimumtact commented:
safe weapons list sucks so :+1:  to what remie said

:+1:  over all to this pull

Nienhaus commented:
10/10

Fayrik commented:
@RemieRichards I believe they check for things that aren't guns as well. I only looked at the Securitron code briefly, and it was a bit above me really.
I'll take another look into it if I do another update to this.
The other issues is that if I do an update, I'll want to add the "hacked" shots again, but there are more issues with that.

RemieRichards commented:
@Fayrik the one thing that's not a gun they check for is a telescopic baton.
EVERYTHING they check for is atleast ```/weapon```, so the ```ignoreThreatAssess``` var would go on ```/weapon```

dannno commented:
haha cool

Atskadan commented:
wait so does cargo get foam guns by default or through a crate

Fayrik commented:
@Atskadan There's two orderable crates (one's contraband, to make it extra fun), and an autolathe recipe for the ammo.
Then the crossbow is still obtainable by arcade machine, and the SMG is admin only. For now.

RandomMarine commented on code/modules/projectiles/ammunition/ammo_casings.dm:
```diff
@@ -199,8 +203,22 @@
+
+/obj/item/ammo_casing/caseless/foam_dart/riot
+	name = "riot foam dart"
+	desc = "Who's smart idea was it to use toys as crowd control? Ages 18 and up."
```
Poor aged 17 spacemen.

optimumtact replied:
Only 18 and up allowed on server woop woop, although that's a very /tg/ specific rule

RemieRichards replied:
@optimumtact I think he's joking about how the lowest age you can give your ingame character is 17.

Lumipharon commented:
New sec meta: Checking records and arresting 17 year old kiddies with foam force guns.

WJohn commented:
This is pretty adorable. Are the magazines shared with regular guns? Can they accept both kinds of ammo? Or are these unique yet identical looking?

Incoming5643 commented:
the colored muzzles are a nice touch

Fayrik commented:
@Lumipharon It's a legal grey area for a reason!

@WJohn GUNCODE WOO. One of the hidden purposes for this was to allow me to spawn non lethal guns for people who prayed for them. There are even some limited var-editing possibilities included here, for badminning.

@Incoming5643 I'm not much of a spriter, but thanks!

Boredone commented:
Why does a pistol (with 2 magazines) cost 4 times as much as 7 shotguns? Seems a bit off to me.
Also, it seems on a glance like Autolathes are the only source of foam darts at the moment. Maybe toss a box of them in the toy list?

Miauw62 commented:
this is great and has my full support
:+1:

On Sat, Mar 21, 2015 at 7:24 AM, Boredone <notifications@github.com> wrote:

> Why does a pistol (with 2 magazines) cost 4 times as much as 7 shotguns?
> Seems a bit off to me.
> Also, it seems on a glance like Autolathes are the only source of foam
> darts at the moment. Maybe toss a box of them in the toy list?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/8451#issuecomment-84265590>
> .
>

Fayrik commented:
@Boredone A pistol can hold twice as much ammo, and fire in a semi-automatic mode. It also fits in your pocket.
With live fire ammunition, you have to consider how much damage you do with each shot, but since these guns all share an ammo type, the semi-automatic option is objectively superior every time.
If anyone can give me a reason that shotguns are worth more than just an aesthetic backpack replacement, that occasionally fire foam darts, I'll raise the price in line with that.
(Also, you do get **two** pistols in that crate, not just one.)
I'd love to make boxes of darts more available, and I did consider the arcades as an option. Unfortunately however, the arcade list is dictated by the RNG, and getting two crossbows is far superior to getting two boxes of ammo, with nothing to fire them.
Although I've yet to see these being used for real mock fire-fights, I suspect that boxes of ammunition won't become truly relevant until cargo starts ordering shotguns anyway.

...There's so much meta-game involved in making toy guns. whoa.

Jordie0608 commented:
Death to toy crossbow snowflake :+1:

Fayrik committed:
Adds everything and makes it all work perfectly. It's like I'm almost good at this or something. (Seriously, points of interest: Secbots react to being shot. Dart rigging. Donksoft guns.)

Fayrik commented:
>Secbots react to being shot.
>Shooting them with pens gets you arrested.
>You can now shoot pens from foam dart guns.
This one commit is like all my dreams for this PR come true.

Lumipharon commented:
FOAM OPS
Syndie borg with foam force gun when?

Fayrik committed:
Riot darts are now available to Nuke ops and Traitors. And I am SO done with this.

Fayrik commented:
This is the last commit, I swear. Unless there's a bug, or issue in all of this, I'm not going to touch it again.
So that's it. Quite a  few good suggestions didn't get in, but for the most part, I've set out what I intended to complete, along with some massive extensions to that.
So get reviewing or whatever, because this PR is complete, and my work here is done.

cookingboy3 commented:
:+1: waiting for next PR with foam equipped Gygax

Fayrik commented:
@cookingboy3 / @Lumipharon I considered borgs and mechs briefly before making this PR. I've decided against adding that as a direct feature, but now the foundation is in, if a group of ops are running a non lethal round for real and have a borg or mech, any coders/admins online should be able to modify the existing guns to be foam-y equivalents.

MrPerson commented:
:+1: to making the toy guns actual guns

:-1: to giving nuke ops toy guns. It's an objectively shitty option both from a gamist (why stun when you can kill for almost the same price?) and roleplay pov (the guys attempting to blow up the station with everyone on it don't want to kill anyone? what do you think the nuke is going to do!). The cheap one is just terrible because a real gun is only 2 telecrystals more. The saw is almost worth it but even still, it's pretty awful. 20 stamina damage != 60 brute. Sure it's a lot cheaper but if you care about price, just go full cheap and get a c20-r. It's even cheaper AND would be way more effective in pretty much all cases.

Fayrik commented:
The toy c20-r was almost just 10 tc each, but that then makes it cost exactly as much as the pistol.
Either way, these aren't really meant to be part of a regular line up, so having them at a higher price is going to discourage their use in the average round.

optimumtact commented:
People can pick a terrible choice if they want.

Giacom commented:
>Carrying a toy gun will cause Securitrons with weapons check enabled to arrest you.

This is what the orange cap on the tip of the gun is for!

Miauw62 commented:
Image recognition is hard, gia
Op 23-mrt.-2015 16:00 schreef "Alex" <notifications@github.com>:

> Carrying a toy gun will cause Securitrons with weapons check enabled to
> arrest you.
>
> This is what the orange cap on the tip of the gun is for!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/8451#issuecomment-85037847>
> .
>

optimumtact commented:
We have AI

Fayrik commented:
Yeah, updated the post to reflect the fact that I removed their permit requiring status when I refactored the security bots.
As funny as I thought it was, I can imagine it could potentially ruin a good moment so yeah.
Easily modified though, if anyone changes their minds on the subject.

optimumtact commented:
:+1: ready for merge

RandomMarine commented:
>>Carrying a toy gun will cause Securitrons with weapons check enabled to arrest you.

>This is what the orange cap on the tip of the gun is for!

Blue crayon making your guns beepsky-proof when?
